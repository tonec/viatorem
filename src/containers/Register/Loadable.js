import Loadable from 'react-loadable'
import { Loading } from 'components'

let RegisterLoadable = Loadable({
  loader: () => import('./Register'),
  loading: Loading
})

export default RegisterLoadable
