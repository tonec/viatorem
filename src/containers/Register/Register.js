import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { register } from 'redux/modules/auth/actions'
import { OneColumnThin } from 'components'
import RegisterForm from 'containers/Register/RegisterForm/RegisterForm'

export class Register extends Component {

  static propTypes = {
    register: PropTypes.func.isRequired
  }

  register = data => {
    return this.props.register(data)
  }

  render () {
    return (
      <OneColumnThin title="Register">
        <RegisterForm onSubmit={this.register} />
      </OneColumnThin>
    )
  }
}

export default connect(() => ({}), { register })(Register)
