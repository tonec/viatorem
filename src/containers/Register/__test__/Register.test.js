import React from 'react'
import { shallow } from 'enzyme'
import { Register } from '../Register'

describe('Register', () => {
  let mockRegister

  beforeEach(() => {
    mockRegister = jest.fn()
  })

  it('renders according to snapshot when enabled', () => {
    const component = shallow(<Register register={mockRegister} />)
    expect(component).toMatchSnapshot()
  })

  it('should call the register function when form submitted', () => {
    const component = shallow(<Register register={mockRegister} />)
    component.find('ReduxForm').simulate('submit')
    expect(mockRegister.mock.calls).toHaveLength(1)
  })
})
