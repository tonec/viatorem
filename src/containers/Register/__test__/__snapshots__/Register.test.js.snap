// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register renders according to snapshot when enabled 1`] = `
<OneColumnThin
  className=""
  offset={9}
  span={6}
  title="Register"
>
  <ReduxForm
    onSubmit={[Function]}
  />
</OneColumnThin>
`;
